plugins/redis/redis.cpp

```diff
@@ -375,10 +473,250 @@ void Connection::parseOptions(ICodeContext * ctx, const char * _options)
+     * To solve these two (client) issues both buffers are cleared before unsubscribing. However, the same applies for the server side output buffer and more importantly the
+     * client socket file descriptor as the redis server pushes published messages to clients. In order to reuse the connection for subsequent subscriptions this unsubscribe
+     * must be confirmed and in order to do this all (now) spurious preceding replies must first be consumed. Because a single socket read will read more than a single
+     * redis reply (if >1 exists waiting on the server), we can switch subsequent reads to just the input buffer. The point here is that we want to limit the number of
+     * possible socket reads and thus possible waits. We then only make a maximum of HPCC_REDIS_PLUGIN_UNSUBSCRIBE_READ_ATTEMPTS socket reads.
+     */
+
+    if (!subscribed)
+        return;
+
+    //return early
+    if (!context || (context->err != REDIS_OK) || channel.isEmpty())
+    {
+        freeContext();
+        return;
+    }
+
+    /* hiredis v0.12.0 is broken (hiredis.h includes read.h but read.h is not installed), and <=v0.11.0 sds.h is neither installed nor included in hiredis.h.
+     *
+     * SubConnections are only used to sub and unsub. They only sub to a single channel at a time (currently) so only an un-flushed UNSUBSCRIBE may still be
+     * still be present in the output buffer. If so it will either be for the same channel or another and thus is benign in nature as long as the channel is
+     * is also confirmed in the reply. This implies that it should be ok to not clear the output buffer. Further more, connection caching is only used for
+     * hiredis >= v0.13.0 which is where this is preferably needed.
+     */
+#if HIREDIS_MAJOR >= 0 && 12 >= MIN_HIREDIS_MINOR && HIREDIS_PATCH >= 1
+    //Empty output buffer as it may contain previous and now unwanted commands. Since the subscription connections only sub and unsub, this should already be empty.
+    if (*context->obuf != '\0')//obuf is a redis sds string (chr*) containing a header with the actual pointer pointing directly to string buffer post header.
+    {
+        sdsfree(context->obuf);//free/clear current buffer
+        context->obuf = sdsempty();//setup new one ready for writing
+    }
+#endif
+
+    //Empty input buffer as it may contain previous and now unwanted replies
+    if (context->reader->len > 0)
+    {
+        redisReaderFree(context->reader);
+        context->reader = redisReaderCreate();
+    }
+
+    //Write command to buffer, set timeout, and write to socket.
+    bool cmdAppendOK = redisAppendCommand(context, "UNSUBSCRIBE %b", channel.str(), channel.length());
+    if ((cmdAppendOK != REDIS_OK) || (redisSetUnsubscribeTimeout() != REDIS_OK) || (writeBufferToSocket() != REDIS_OK))
+    {
+        freeContext();
+        return;
+    }
+
+    getUnsubscribeReadAttempts();
```

Comment:
This should use the value returned from the function.  If you did want to use the member variable instead (not my preferred option) you should not return a value from the function, and call it something else that doesn't begin with get.